legal-api/src/legal_api/resources/v2/business/business_tasks.py

```diff
@@ -104,21 +105,22 @@ def construct_task_list(legal_entity):  # pylint: disable=too-many-locals; only
     entity_types_no_ar = ["SP", "GP"]
     tasks = []
     order = 1
+    business_id = business.id if business.is_legal_entity else business.legal_entity_id
 
-    warnings = check_warnings(legal_entity)
+    warnings = check_warnings(business)
     if any(x["warningType"] == WarningType.MISSING_REQUIRED_BUSINESS_INFO for x in warnings):
         # TODO remove compliance warning line when UI has been integrated to use warnings instead of complianceWarnings
-        legal_entity.compliance_warnings = warnings
-        legal_entity.warnings = warnings
+        business.compliance_warnings = warnings
+        business.warnings = warnings
 
         # Checking for draft or pending conversion
-        if not Filing.get_incomplete_filings_by_type(legal_entity.id, "conversion"):
-            tasks.append(create_conversion_filing_todo(legal_entity, order, True))
+        if not Filing.get_incomplete_filings_by_type(business_id, "conversion"):
+            tasks.append(create_conversion_filing_todo(business, order, True))
             order += 1
 
     # Retrieve filings that are either incomplete, or drafts
     pending_filings = Filing.get_filings_by_status(
-        legal_entity.id,
+        business_id,
```

Comment:
as per comment above, needs to work with business instance instead of business.id

Reply:
updated